### PR TITLE
[#322] Extract contribution guide from readme to github wiki

### LINKED ISSUE #322: Create template wiki
## Why

Reorganize content in `README.md` to wiki pages. 

- On `README.md`, we can simply have the instruction to use the template.
- Other content, e.g. contribution guide can go into the wiki
- Adding some more content in the wiki regarding common issues and a guide to test the template locally
 
## Who Benefits?

Developers who work on the template

### PR DESCRIPTION
**Pull request by mosharaf13**
close #322

## What happened 👀

Remove the contribution guide from readme and add it as a section in GitHub Wiki.

## Insight 📝

The contribution guide should appear as a link in the sidebar of GitHub Wiki.

## Proof Of Work 📹

Screen Shot 2023-07-06 at 6 58 27 PM

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,124 +62,6 @@ make dev
...
-- `SCSS`
-- `SHELL`
-
-For other files that are not fit the types above, use the extension as the name
-e.g. `TOOL_VERSION` for `.tool-version` file.
-
-For the normal string, name it after the content
-e.g. `ERROR` for template error message.
-
 ## Testing the Template
```
Comment (malparty) [inline README.md:65]: This part too can be moved in the wiki :)

File: README.md [hunk 2]
```diff
@@ -62,124 +62,6 @@ make dev
 
 Read more about Rails Application Template in the [official Rails Guides](https://guides.rubyonrails.org/rails_application_templates.html).
```
Comment (malparty) [inline README.md:64]:
```suggestion

See the contribution guidelines in the [Wiki](../../wiki).
```

Should we keep a small mention of the wiki in the ReadMe file? 🤔

**Review by malparty — CHANGES_REQUESTED** [review]
Just minor suggestions, the rest is nice :)

File: README.md (hunk 1 above)
Comment (mosharaf13) [reply README.md:65]: Do you mean the `Testing the template` section?

File: README.md (hunk 2 above)
Comment (mosharaf13) [reply README.md:64]: Thanks 🙏🏼 . Fixed in 45ceeed0eaa8e94885f7135afe170258845f71a1

File: README.md (hunk 1 above)
Comment (malparty) [reply README.md:67]: Yes, as suggested in the issue :)

> On README.md, we can simply have the instruction to use the template.

But I agree it was not really explicit 🙈

File: .template/addons/github/.github/wiki/_Sidebar.md [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 - [[Home]]
 - [[Getting Started]]
+- [[Contribution_Guide]]
```
Comment (sanG-github) [inline .template/addons/github/.github/wiki/_Sidebar.md:5]: Should we use it without the `underscore`, it's more readable indeed!!!

```suggestion
- [[Contribution Guide]]
```

Comment (sanG-github) [reply .template/addons/github/.github/wiki/_Sidebar.md:5]: Please also update the tests. Thanks!!!

**Review by sanG-github — APPROVED** [review]
Approved with a minor suggestion! 🚀

File: .template/addons/github/.github/wiki/_Sidebar.md (hunk 3 above)
Comment (mosharaf13) [reply .template/addons/github/.github/wiki/_Sidebar.md]: Thanks. Fixed in f6a068eaa9bd57a1bdccf34b313b9b56fd7d7290

File: README.md (hunk 1 above)
Comment (mosharaf13) [reply README.md]: I have extracted testing section from readme to wiki. However, [here](https://github.com/nimblehq/rails-templates/actions/runs/5505936136/jobs/10033947370) the test generated app workflow is failing on some tests related to `devise` gem. 

I checked on generated app, and saw that devise comes as add-on. My question is, are those tests supposed to be there or are they outdated? 

N.B. I made changes to readme and wiki pages only. So, failure in devise related tests is a bit unexpected 😕

File: .template/addons/github/.github/wiki/Testing.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+```sh
+.template/bin/rubocop
+```
+
+Any RuboCop command options can be passed:
+
+
+```sh
+# Run Ru`boCop with auto correct
```
Comment (tyrro) [inline .template/addons/github/.github/wiki/Testing.md:14]:
```suggestion
# Run RuboCop with auto correct
```

File: .template/addons/github/.github/wiki/_Sidebar.md [hunk 5]
```diff
@@ -2,6 +2,7 @@
 
 - [[Home]]
 - [[Getting Started]]
+- [[Contribution Guide]]
```
Comment (malparty) [inline .template/addons/github/.github/wiki/_Sidebar.md:6]:
```suggestion
- [[Testing]]

```

Then we can add testing too 😇

**Review by malparty — APPROVED** [review]
Approved ahead with a last comment and Sang/Rajib's comments :)

File: .template/addons/github/.github/wiki/_Sidebar.md (hunk 5 above)
Comment (mosharaf13) [reply .template/addons/github/.github/wiki/_Sidebar.md:6]: Testing is linked under `Operations` section. Should it also be added here?

File: .template/spec/addons/base/github/template_spec.rb [hunk 6]
```diff
@@ -21,9 +21,7 @@
 
   # TODO: Can't test this as it is now ignored by `.dockerignore`
   xit 'modifies the README.md' do
-    expect(file('README.md')).to contain('## Documentation')
-
-    expect(file('README.md')).not_to contain('## Getting Started')
+    expect(file('README.md')).not_to contain('## Get Started')
```
Comment (tyrro) [inline .template/spec/addons/base/github/template_spec.rb:24]: Is this change needed?  🤔

File: .template/spec/addons/base/github/template_spec.rb [hunk 7]
```diff
@@ -67,11 +65,11 @@
     it 'exists' do
       expect(file('.github/wiki/Testing.md')).to exist
     end
+  end
 
-    it 'contains the correct content extracted from README.md' do
-      expect(file('.github/wiki/Testing.md')).to contain('### Docker-based tests on the CI server')
-      expect(file('.github/wiki/Testing.md')).to contain('### Test')
-      expect(file('.github/wiki/Testing.md')).to contain('### Automated Code Review Setup')
+  describe '.github/wiki/Contribution-Guide.md' do
```
Comment (tyrro) [inline .template/spec/addons/base/github/template_spec.rb:70]: I think this `describe` block needs to be inside a separate block, just like `describe '.github/wiki/Authentication.md'` or `describe '.github/wiki/CI-CD.md'`

**Review by tyrro — CHANGES_REQUESTED** [review]
Also, please have a look at the failed tests when you have time!

File: .template/spec/addons/base/github/template_spec.rb (hunk 6 above)
Comment (mosharaf13) [reply .template/spec/addons/base/github/template_spec.rb]: Thanks 👍🏼 . Reverted in c98430ab389ec98d650d02182b2fd473096f7529

File: .template/addons/github/.github/wiki/_Sidebar.md (hunk 5 above)
Comment (malparty) [reply .template/addons/github/.github/wiki/_Sidebar.md:6]: Yes, as this is a root page of the wiki ;-)

File: README.md [hunk 8]
```diff
@@ -62,138 +62,9 @@ make dev
...
 
-## How to contribute
+See the contribution guidelines in the [Wiki](../../wiki/Contribution_Guide).
```
Comment (olivierobert) [inline README.md:65]: How about adding a heading above these links:

```
## Contribution Resources

- [Contributions Guidelines](../../wiki/Contribution_Guide).
- [Testing the template](../../wiki/Testing)
```

Comment (mosharaf13) [reply README.md]: Yeah. This looks much cleaner 👍🏼. Thanks. Fixed in 4415e12bdcfae4b0daa7ea1835d69fb1fe7515ab

**Review by olivierobert — APPROVED** [review]

**Comment by tyrro** [comment]
@mosharaf13 ,  I see a few [tests are failing ](https://github.com/nimblehq/rails-templates/actions/runs/5595430962/job/15155989784?pr=438) in the devise add-on section, are you facing the same in development?

**Comment by mosharaf13** [comment]
> tests are failing

No. When I generate project from the template `devise` gem is in correct location of the gemfile. However, for some reason, failed tests saying it's not getting 'devise' gem in correct location.

N.B. I made changes to readme and wiki pages only.

**Comment by malparty** [comment]
@mosharaf13 maybe a simple rebase on the latest version of develop can help fix the tests issue? 💭

**Comment by mosharaf13** [comment]
> @mosharaf13 maybe a simple rebase on the latest version of develop can help fix the tests issue? 💭 

@malparty I have submitted my laptop already. I don't have any laptops with me right now. I will try rebasing once I get hold of a laptop.

**Comment by malparty** [comment]
Okay, I finally found why the tests where failing :harold:

=> The wiki was created inside the generated application wiki, instead of at the root of this repository ^^

**Review by tyrro — APPROVED** [review]
